backend/src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,35 @@
+import { Request, Response } from 'express';
+import { createUser } from '../services/user.service';
+import { getUserTokens, extractToken } from '../services/auth.service';
+import User from '../interfaces/User';
+import jwt from 'jsonwebtoken';
+
+export const register = async (req: Request, res: Response) => {
+  try {
+    const { userName, email, password } = req.body;
+    if (!userName || !email || !password) {
+      return res
+        .status(400)
+        .json({ message: 'Please provide all required fields' });
+    }
+    const user: User = { userName, email, password };
+    const newUser = await createUser(user);
+    res.status(201).json(newUser);
+  } catch (error: any) {
```

Comment:
this has to be `any` type because I cant control the type that will be thrown as an error, the compiler tells me I have to use the `any` or `unknown` types here